doc/developer-guides/hld/ivshmem-hld.rst

```diff
@@ -0,0 +1,97 @@
+.. _ivshmem-hld:
+
+ACRN Shared Memory Based Inter-VM Communication
+###############################################
+
+To support inter virtual machines communication based on shared memory
+mechanism, ACRN device model/hypervisor emulate a virtual PCI device
+(Called ivshmem device) to expose the shared memory base address and
+size. Currently, ACRN only supports post-launched to post-launched
+communication via ivshmem device. In future, ACRN will supports
+pre-launched VM inter-VM communication.
+
+Inter-VM Communication Overview
+*******************************
+
+.. figure:: images/ivshmem-image1.png
+   :align: center
+   :name: ivshmem-architecture-overview
+
+   ACRN Inter-VM Communication Architecture Based on Shared Memory
+
+Two solutions for ivshmem device implementation, one is called DM-land,
+another is called Hypervisor-land. For DM-land, the ivshmem device is
+emulated in device model and the inter-VM communication shared memory
+region is allocated from Service memory. For Hypervisor-land, the
+ivshmem device is emulated in hypervisor and the shared memory is
+allocated from hypervisor. The former only work for post-launched VMs,
+the latter can work for both of pre-launched and post-launched VMs. The
+Hypervisor-land will be implemented in the future.
```

Comment:
Two solutions of ivshmem device: DM-land and hypervisor-land.  For DM-land, the ivshmem device emulated in acrn device model and its shared memory region is allocated from SOS's memory space. For hypervisor-land, it emulated in hypervisor and the shared memory regions are reserved in hypervisor's memory space. The former solution only support interactive between port-launched VMs and the latter one can work for both pre-launched and post-launched VMs. The hypervisor-land solution is WIP, plan to support in the near future.